frontend/src/components/SaveTranscript/SaveModal.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { db } from '../../../../backend/firebase/firebase.config.js';
+import { doc, updateDoc, arrayUnion, getDoc, setDoc, collection, addDoc } from 'firebase/firestore';
+import { useAuth } from '../auth/auth.jsx';
+
+const SaveModal = ({ isOpen, onClose, onSave }) => {
+    const [transcriptName, setTranscriptName] = useState('');
+    const folderNameRef = useRef(null);
+    const [error, setError] = useState('');
+    const [folders, setFolders] = useState([])
+    const [selectedFolder, setSelectedFolder] = useState(folders[0] || '');
+    const { currentUser } = useAuth();
+
+
+    if (!isOpen) return null;
+
+
+        const fetchFolders = async () => {
+            const userDocRef = doc(db, "users", currentUser.uid);
+            const userDoc = await getDoc(userDocRef);
+
+            if (userDoc.exists()) {
+                const userData = userDoc.data();
+                setFolders(userData.folderNames || [])
+            }
+        }
+        fetchFolders();
+
+    const handleNewFolder = async (e) => {
```

Comment:
can we combine with the earlier "handleNewFolder" function? lets avoid copy & paste if possible